### ISSUE TITLE
How to load fixtures using Alice once before all tests?

### ISSUE THREAD

**Issue by Kwadz** [liip/LiipFunctionalTestBundle#303]
How to [load fixtures using Alice](https://github.com/liip/LiipFunctionalTestBundle#loading-fixtures-using-alice) once before all tests?

**Comment by Kwadz** [comment]
Is there at least a workaround currently? Did you already plan to implement something? Let's talk about a base to start.

**Comment by alexislefebvre** [comment]
The only workaround I can think today is to use `setUp()` in order to load fixtures only one per class.

**Comment by Kwadz** [comment]
You can't use `setUp()` which is invoked before each test. Using `__construct()` only allows to load fixtures before the tests of the given class.  

May be we could find a way to execute `Liip\FunctionalTestBundle\Test\WebTestCase::loadFixtureFiles()` from a PHPUnit listener:

```php
class TestSuiteListener extends PHPUnit_Framework_BaseTestListener
{
    public function startTestSuite(\PHPUnit_Framework_TestSuite $suite)
    {
        printf("TestSuite '%s' started.\n", $suite->getName());
        // load fixtures...
    }
}
```

What do you think?

**Comment by alexislefebvre** [comment]
I didn't know PHPUnit listeners, it seems to be  interesting.

It may be possible by defining a trait that load fixtures and is loaded by `setUp()` in each test class. In the same time it should declare a constant so that fixtures would be loaded only once.

Anyway, your question may have more visibility on StackOverflow. If I remember correctly, there is a tag about this Bundle.

**Comment by Kwadz** [comment]
I think `LiipFunctionalTestBundle` should suggest an official way in the documentation, as its purpose is helping functional/integration testing. I am also going to do some research on my side to find the "best" way.

**Comment by Kwadz** [comment]
We can load all fixtures once before all tests by [customizing the Symfony bootstrap process](https://symfony.com/doc/current/testing/bootstrap.html).

**Comment by alexislefebvre** [comment]
If you manage to load fixtures only once by modifying the bootstrap process, I think that it would be interesting to add it in this Bundle's configuration.

**Comment by jewome62** [comment]
@Kwadz Ok, but the (fixture) reference repository is not available into the tests ?

Thanks

**Comment by Kwadz** [comment]
@jewome62 Sorry I didn't understand well your question, can you rephrase it with more details please?

**Comment by jewome62** [comment]
Hello,

If I use the load fixture in php unit bootstrap.
I can't have the reference repository into test file.

And I can t use the fixture reference into my test ???

Thanks
